File: app/src/main/java/ai/elimu/analytics/entity/WordLearningEvent.java
package ai.elimu.analytics.entity;

import androidx.annotation.NonNull;
import androidx.room.Entity;

import ai.elimu.model.v2.enums.analytics.LearningEventType;

@Entity
public class WordLearningEvent extends LearningEvent {

    private Long wordId;

    @NonNull
    private String wordText;

    @NonNull
    private LearningEventType learningEventType;

    public Long getWordId() {
        return wordId;
    }

    public void setWordId(Long wordId) {
        this.wordId = wordId;
    }

    public String getWordText() {
        return wordText;
    }

    public void setWordText(String wordText) {
        this.wordText = wordText;
    }

    public LearningEventType getLearningEventType() {
        return learningEventType;
    }

    public void setLearningEventType(LearningEventType learningEventType) {
        this.learningEventType = learningEventType;
    }
}

File: app/src/main/java/ai/elimu/analytics/entity/WordLearningEvent.kt
package ai.elimu.analytics.entity

import ai.elimu.model.v2.enums.analytics.LearningEventType
import androidx.room.Entity

/**
 * Based on https://github.com/elimu-ai/webapp/blob/main/src/main/java/ai/elimu/entity/analytics/WordLearningEvent.java
 */
@Entity
class WordLearningEvent : LearningEvent() {
    lateinit var wordText: String

    /**
     * This field might not be included, e.g. if the learning was done in a 3rd-party app that did
     * not load the content from the elimu.ai Content Provider. In that case, this field will be {@code null}.
     */
    var wordId: Long? = null

    @Deprecated("Will be replaced by `additionalData`")
    var learningEventType: LearningEventType? = null
}
